DocumentManager.js

```diff
@@ -9,21 +9,52 @@ module.exports = class {
         };
     }
 
+    /**
+   * Recursively appends docIds via pagination
+   */
+    async recurseThroughPagination(cursor, docIds) {
+        let response;
+        try {
+            response = await axios.post(
+                "https://api.dropboxapi.com/2/paper/docs/list/continue",
+                {
+                    cursor: cursor
+                },
+                {
+                    headers: this.baseHeaders
+                }
+            );
+        } catch (e) {
+            console.error(e);
+            return docIds;
+        } finally {
+            Array.prototype.push.apply(docIds, response.data.doc_ids);
+            if (response.data.has_more) {
+                await this.recurseThroughPagination(response.data.cursor.value, docIds);
```

Comment:
I understand that `recurseThroughPagination()` here is modifying the `docIds` variable, but I'd _really_ like to be explicit about returning something here. It'd be a lot easier to cognitively understand what's going on, and much more intentional about what this method actually gives you (which is more document IDs). As such, what about this?

```suggestion
                return await this.recurseThroughPagination(response.data.cursor.value, docIds);
```

And then, below that `if` block, add a `return docIds;`. This way, no matter what, the `recurseThroughPagination()` method is always returning document IDs.

Reply:
Absolutely.